### ISSUE TITLE
Seeing ads with uBlock Origin installed...

### ISSUE THREAD

**Issue by sphinx9** [gorhill/uBlock#1483]
I can still see most ads with uBlock Origin installed. There were also ads before YouTube videos but they seem to have stopped (for no reason). Using latest Chrome and uBlock Origin (49 and v1.6.4).

https://i.gyazo.com/e3527d496300f03a8858c9e2d3248e52.png

There are lots of ads that ARE being blocked, but lots that aren't.

Filter lists: https://i.gyazo.com/5be3d25b8299b2fd8aa02f4d8c329390.png

**Comment by Fenslett** [comment]
uBlock filters are made up of filename, directory name, domain name and DOM attribute(i.e, id and class) containing the string such as 'ad' and 'adv', which means advertisements always appear unless filename has such keywords. So if you found an ad uBlock doesn't block, please report or block it using element picker.

**Comment by sphinx9** [comment]
If you can't tell from the screenshot, those are adverts on Google search results. Are you seriously telling me that uBlock is 'unaware' that there are adverts there and I have to report them or block them myself? There is clearly something wrong with the extension that is causing those elements to still appear ^^

**Comment by Hrxn** [comment]
Try the default filter settings.. I don't see those ads. Also, you need to provide steps to reproduce..

**Comment by sphinx9** [comment]
"Also, you need to provide steps to reproduce.."

This has been happening since I installed the extension a few days ago, so steps to reproduce are install the extension then browse the internet, I guess!

**Comment by RoxKilly** [comment]
In your 2nd screenshot, I notice that although `EasyList` is selected, its rules are not being applied (it shows "0 used out of 24,191"). This may be the cause of your problem (EasyList is the most expansive adblock list).  Likewise, `EasyPrivacy` is only using 5 out of 11,149 rules.   Try to hit "Purge all Caches"  and then "Update Now" above the filter lists

**Comment by Hrxn** [comment]
The screenshot with the filter lists does not show the default filter list settings..

**Comment by sphinx9** [comment]
> Try to hit "Purge all Caches" and then "Update Now" above the filter lists

Thanks. I just did this and though the filters list now shows it is using ' 24,102 used out of 24,153' (and similar numbers for the others), that advert in particular still appears.

**Comment by Fenslett** [comment]
@sphinx9 Don't get me wrong. An ad with a weird name or attribute like *path/Z11sDgBfg.gif* cannot be blocked until someone put it on the list. By way of illustration, many advertisers attempt to pass through ad-blocking extensions. Could you tell me the URL?

**Comment by RoxKilly** [comment]
Did the number of rules used also go up for the other filters I mentioned in my last post?  Ok, now try to hold down the SHIFT button while you do a page Reload in the browser.  Same result?

**Comment by sphinx9** [comment]
>  @Fenslett An ad with a weird name or attribute like path/Z11sDgBfg.gif cannot be blocked until someone put it on the list.

Yeah, they're usually in containers like .ad-container or something, and even when not there seem to be ways of hiding them despite their identifiers being obfuscated as other extensions (like adblock/adblock plus/etc) are able to do it. And if it's being obfuscated by being named with random characters then adding it to a list won't do very much.

> @RoxKilly Did the number of rules used also go up for the other filters I mentioned in my last post? Ok, now try to hold down the SHIFT button while you do a page Reload in the browser. Same result?

The number of rules for the others did go up. The ads seem to have disappeared after a hard refresh - thanks very much for your help. :)

Do you (or anyone else) happen to know why or how this has happened? After a week or so of using this extension, I've found it quite unreliable (which is surprising as it's quite esteemed from what I've read online) - things like the aforementioned issue, YouTube video ads playing, etc etc...

**Comment by RoxKilly** [comment]
You're welcome.  I've had very few problems with uBO. It has become my favorite extension.  Others who know more about how the extension stores and reads rules would be better placed than I, but my guess is that you have a non-standard setup.  Maybe you have other security software (a sandbox comes to mind) that interferes with uBO's ability to either save or retrieve its rules.

**Comment by sphinx9** [comment]
The only sandbox I know of is the one inherent in Google Chrome. I don't have anything particularly special or 'weird' going on - Win10, Avast (with web filters / anything web related disabled). Chrome extensions include FoxyProxy (currently disabled), Stylish (simply injects user-defined css (without removing any, afaik)), uBO and that's it really. Was using Disconnect but I disabled it as it's not needed with uBO, it seems.

**Comment by RoxKilly** [comment]
Yeah sounds pretty vanilla.  Sorry I don't know.

**Comment by sphinx9** [comment]
No problem. Thanks for your help. :)

**Comment by Fenslett** [comment]
@sphinx9 Hmm. I've read your issues again since I felt like something seemed wrong during conversation. Sorry for misunderstanding your words from the beginning. Couldn't see the word 'most' in the first sentence. I'm a bit shameful :(

.

> @sphinx9 Do you (or anyone else) happen to know why or how this has happened? After a week or so of using this extension, I've found it quite unreliable (which is surprising as it's quite esteemed from what I've read online) - things like the aforementioned issue, YouTube video ads playing, etc etc...

Browser caching goes awry infrequently such as showing out-of-date contents when the browser doesn't recognize it as a significant change. Whenever it happened, we need to clear the entire browser cache or detour our cache system. (I don't recommend clearing cache because all auto-login credentials are also removed)

Here's how:

**On Windows,**
1. Hold `Ctrl` and press the `F5` (Works for most browsers)
2. Hold `Ctrl` and click *reload* button
3. Hold `Shift` and press the `F5` / Hold `Shift` and click *reload* button

**On Mac,**
1. Hold down `Shift` and click *reload* button
2. Hold both `Cmd` and Shift, and then press `R`

**Comment by sphinx9** [comment]
No problem! Perhaps I didn't explain well enough - currently running on no sleep so my brain's a little slow. :D

Thanks for your suggestion, the issue *seems* to have stopped (for now, at least), but if it recurs as it has done in the past then I will try clearing my browser cache / deleting history etc. (I don't mind re-logging in to things). :)

**Comment by IDKwhattoputhere** [comment]
You deselected "Parse and enforce cosmetic filters.". Don't be surprised if ads show up then. In the future try [this](https://cloud.githubusercontent.com/assets/16030872/13845989/0c2e099a-ec44-11e5-8da5-5b39a9a118b2.png) first before posting.

**Comment by gorhill** [comment]
> You deselected "Parse and enforce cosmetic filters."

**Comment by sphinx9** [comment]
Sorry but that's another issue and one that hasn't really been addressed. If deselecting that option is the cause of the problems I have been having then why is it only in some places, and intermittently at that? Also, if the ambiguously labelled "parse and enforce cosmetic filters" option does that then what does the extension even do with that option disabled?

**Comment by IDKwhattoputhere** [comment]
There are two kinds of filters: Cosmetic filters and network filters.
Read this: https://github.com/gorhill/uBlock/wiki/Does-uBlock-block-ads-or-just-hide-them%3F

**Comment by harshanvn** [comment]
> Also, if the ambiguously labelled "parse and enforce cosmetic filters" option does that then what does the extension even do with that option disabled?
   It will still enforce all the other types of filters. It will ignore cosmetic filters. Read this, to get more information - [Overview of uBlock Filtering Engine](https://github.com/gorhill/uBlock/wiki/Overview-of-uBlock's-network-filtering-engine:-details)

**Comment by gorhill** [comment]
@sphinx9 The question to answer: why did you un-select that option?

**Comment by sphinx9** [comment]
@gorhill To be honest I don't remember doing so, but I tried numerous things to try to get the extension to function as intended when I first installed it (see issues previously mentioned in this thread). It doesn't really make sense that there is an option to essentially disable the main functionality of the extension but you've labelled the issue as invalid so I guess this is over?

**Comment by IDKwhattoputhere** [comment]
Read the resource I or the one @harshanvn linked. The option only disables cosmetic filters but all others will still be used.

**Comment by sphinx9** [comment]
Perhaps it's just me but I don't see the point of that. Thanks for your reply.

**Comment by IDKwhattoputhere** [comment]
The point is that with cosmetic filters disabled you will still get a smaller privacy exposure and pages will still load faster all while uBO will use less RAM.

**Comment by gorhill** [comment]
> It doesn't really make sense that there is an option to essentially disable the main functionality of the extension

The explanation of why this option exists is available right there in the UI:

> This option enables the parsing and enforcing of Adblock Plus-compatible “element hiding” filters. These filters are essentially cosmetic, they serve to hide elements in a web page which are deemed to be a visual nuisance, and which can't be blocked by the net request-based filtering engine.
> 
> Enabling this feature increases uBlock₀'s memory footprint.

For less powerful devices, this may be useful.

**Comment by sphinx9** [comment]
Yes, I've read that already whilst trying to get the extension to function as expected (see previous comments in this thread). Thanks for your explanation, however. You can delete this entire thread, by the way. It doesn't serve much purpose.

**Comment by gorhill** [comment]
I would say that there is ground for an issue here: uBO should have counted the cosmetic filters which were discarded -- i.e. EasyList should have shown "24,000 used out of 60,000" -- this at least would have provided one extra hint about the issue. But this needs to be opened in its own dedicated issue.

**Comment by sphinx9** [comment]
I suppose that is true! I didn't even realise that the vast majority of the filters weren't active, so perhaps it'd be a good idea to make it a bit more obvious. (UI design isn't my forté, however...)

**Comment by gorhill** [comment]
@sphinx9 Can you open a new issue specifically for this? Use your 2nd picture, it demonstrates well the issue.

**Comment by sphinx9** [comment]
@gorhill sure, done - https://github.com/gorhill/uBlock/issues/1484

**Comment by gorhill** [comment]
Thanks.
